ext/liquid_c/vm_assembler.c

```diff
@@ -298,15 +354,17 @@ void vm_assembler_add_filter(vm_assembler_t *code, VALUE filter_name, size_t arg
     st_data_t builtin_index;
     bool is_builtin = st_lookup(builtin_filter_table, filter_name, &builtin_index);
 
-    uint8_t *instructions = c_buffer_extend_for_write(&code->instructions, is_builtin ? 3 : 2);
     if (is_builtin) {
+        uint8_t *instructions = c_buffer_extend_for_write(&code->instructions, 3);
         *instructions++ = OP_BUILTIN_FILTER;
         *instructions++ = builtin_index;
+        *instructions++ = arg_count + 1; // include input
```

Comment:
This will be specifically for `built_in` filters.
Built in filter instruction:
```
[OP_FILTER][BUILT_IN_INDEX][ARG_COUNT] # 8 + 8 + 8 bit
```

Filter instruction:
```
[OP_FILTER][Constant Index] # 8 + 16 bit (constant = [filter_name, num_args])
```